srtcore/channel.cpp

```diff
@@ -324,30 +326,40 @@ void CChannel::setIpV6Only(int ipV6Only)
 int CChannel::getIpTTL() const
 {
    socklen_t size = sizeof(m_iIpTTL);
-   if (m_iIPversion == AF_INET)
+   if (m_BindAddr.family() == AF_INET)
    {
       ::getsockopt(m_iSocket, IPPROTO_IP, IP_TTL, (char *)&m_iIpTTL, &size);
    }
-   else
+   else if (m_BindAddr.family() == AF_INET6)
    {
       ::getsockopt(m_iSocket, IPPROTO_IPV6, IPV6_UNICAST_HOPS, (char *)&m_iIpTTL, &size);
    }
+   else
+   {
+       // If family is unspecified, the socket probably doesn't exist.
+       return -1;
+   }
```

Comment:
`SRTO_IPTTL` in API.md: add -1 as a valid return value.

Reply:
That's actually a different problem.

This should have been reported as an exception so that when indirectly called from `srt_getsockflag` it will result with error. This actually shouldn't normally happen (therefore logged as IPE): When not opened, it will return the value as recorded in the internal fields. When opened, the family should already be set, as the socked wouldn't be able to be created without it.